tvplayer/src/main/java/com/kaltura/tvplayer/KalturaPlayer.java

```diff
@@ -85,67 +87,112 @@
     private PlayerTokenResolver tokenResolver = new PlayerTokenResolver();
     private PlayerInitOptions initOptions;
 
-    public static KalturaPlayer createOVPPlayer(Context context, PlayerInitOptions initOptions) {
-        kalturaPlayerType = KalturaPlayerType.ovp;
-        KalturaPlayer kalturaPlayer = new KalturaPlayer(context, initOptions);
-        kalturaPlayer.serverUrl = KalturaPlayer.safeServerUrl(initOptions.serverUrl, KalturaPlayer.DEFAULT_OVP_SERVER_URL);
-        return kalturaPlayer;
+    public static void initializeOVP(Context context, int partnerId, @Nullable String serverUrl) {
```

Comment:
if `initializeOVP` and `initializeOTT` only differ in type and app needs to know it's type to call the correct method then what is the benefit of creating both handlers? just pass the type(that has a nice EUM already) as a parameter.

Reply:
The benefit is that serverUrl is nullable in OVP (since we default to the SaaS) and not nullable in OTT.